### ISSUE TITLE
GHC-7.2 build failure

### ISSUE THREAD

**Issue by bergmark** [kolmodin/binary#81]
```
src/Data/Binary/Put.hs:61:1:
    bytestring-0.9.2.0:Data.ByteString can't be safely imported! The module itself isn't safe.
xcabal: Error: some packages failed to install:
binary-0.7.6.0 failed during the building phase. The exception was:
ExitFailure 1
```

**Comment by kolmodin** [comment]
With Travis CI we check all major GHC versions back to GHC 7.4.2 which was released 3+ years ago.
https://travis-ci.org/kolmodin/binary
Do you need `binary` to work with an older GHC?

**Comment by bergmark** [comment]
Personally I don't, I saw this on the [matrix builder](http://matrix.hackage.haskell.org/package/binary) and thought it may be a mistake in the latest release.

A lot of packages still  support 7.0. Since binary is high up in the hierarchy it would effectively force others to drop support as well. I'd argue that it's worth keeping 7.0+ support unless there is a concrete reason not to.

I'd be happy to send a PR for this by the way.

**Comment by kolmodin** [comment]
Looks like bytestring-0.9.2.0 doesn't provide the Safe Haskell pragmas.
`Data.Binary.Put` has this Safe Haskell pragma.
```hs
#if __GLASGOW_HASKELL__ >= 701
{-# LANGUAGE Safe #-}
#endif
```
Maybe there are other modules that'd fail as well.
Not sure what Safe Haskell guarantees we can provide if bytestring didn't set any pragmas, but yes, we can do better than failing. :)

If you'd like to do a PR that'd be great! Thanks!

**Comment by kolmodin** [comment]
Fixed with #83.
